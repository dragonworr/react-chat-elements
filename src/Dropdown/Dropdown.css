.rce-dropdown{
    max-width: 200px;
    padding: 8px;
    background: #f5f5f5;
    border-radius: 5px;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    transform: scale(0);
    margin: 100px;
    position: relative;
    -webkit-box-shadow: 0px 5px 5px -5px rgba(0,0,0,0.75);
    -moz-box-shadow: 0px 5px 5px -5px rgba(0,0,0,0.75);
    box-shadow: 0px 5px 5px -5px rgba(0,0,0,0.75);
    transform-origin: left top;
    animation: dropdown-scaling 0.5s ease forwards;
}

@keyframes dropdown-scaling{
    0%{
        opacity: 0;
    }

    50%{
        opacity: 0.5;
        transform: scale(1.1);
    }

    100%{
        opacity: 1;
        transform: scale(1);
    }
}

.rce-dropdown.top:before{
    content:'';
    width: 0;
	height: 0;
    border-left: 15px solid transparent;
	border-bottom: 15px solid #f5f5f5;    
    border-right:15px solid transparent;
    border-top:15px solid transparent;
    position: absolute;
    left: 0;
    right: 0;
    top:-25px;
    margin: auto;
}

.rce-dropdown.bottom:before{
    content:'';
    width: 0;
	height: 0;
    border-left: 15px solid transparent;
	border-bottom: 15px solid transparent;    
    border-right:15px solid transparent;
    border-top:15px solid #f5f5f5;
    position: absolute;
    right: 0;
    left:0;
    bottom:-25px;
    margin: auto;
}

.rce-dropdown.left:before{
    content:'';
    width: 0;
	height: 0;
    border-left: 15px solid transparent;
    border-top: 15px solid transparent;   
	border-bottom: 15px solid transparent;    
    border-right:15px solid #f5f5f5;
    position: absolute;
    left: -25px;
    top:0;
    bottom:0;
    margin: auto;
}

.rce-dropdown.right:before{
    content:'';
    width: 0;
	height: 0;
    border-left: 15px solid #f5f5f5;
    border-top: 15px solid transparent;   
	border-bottom: 15px solid transparent;    
    border-right:15px solid transparent;
    position: absolute;
    right: -25px;
    top:0;
    bottom:0;
    margin: auto;
}

.rce-dropdown ul{
    list-style: none;
    margin:0;
    padding: 0;
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    text-align: center;
}

.rce-dropdown ul li{
    color:#767676;
    font-size: 12px;
    padding: 5px 0;
    cursor: pointer;
    width: 100%;
    border-bottom: 1px solid #e9e9e9;
}

.rce-dropdown ul li:last-child{
    border:none;   
}

.rce-dropdown ul li:hover a{
    color:#3a6d8c;
}

.rce-dropdown.fade{
    opacity: 0;
    transform: scale(1);
    animation: dropdown-fade 0.5s ease forwards;
}

@keyframes dropdown-fade{
    0%{
        opacity: 0;
    }

    100%{
        opacity: 1;
    }
}