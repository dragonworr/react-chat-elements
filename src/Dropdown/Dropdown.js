import React, { Component } from 'react';
import './Dropdown.css';

import Button from '../Button/Button';

const classNames = require('classnames');

export class Dropdown extends Component {
	constructor(props) {
		super(props);

		this.state = {
			show: false,
		};
	}

	onBlur(e) {
		if (this.state.show === true)
			this.setState({show: false});
	}

	render() {
		return (
			<div className='rce-dropdown-container' onBlur={this.onBlur.bind(this)}>
				{
					<Button
						{...this.props.buttonProps}
						onClick={() => this.setState({show: !this.state.show})}/>
				}
				{
					this.state.show &&
					<div
						className={
							classNames(
								'rce-dropdown',
								this.props.animationType,
								'rce-dropdown-open__' + this.props.animationPosition,
							)
						}>
						<ul>
							{
								this.props.items.map((x, i) => (
									<li key={i} onMouseDown={(e) => this.props.onSelect(i)}>
										<a>{x}</a>
									</li>
								))
							}
						</ul>
					</div>
				}
			</div>
		);
	}
}

Dropdown.defaultProps = {
	animationType: 'default',
	animationPosition: 'nortwest',
	items: [],
	onSelect: Function,
	buttonProps: null
};

export default Dropdown;
