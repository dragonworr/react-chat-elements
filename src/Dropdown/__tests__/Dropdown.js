import React, { Component } from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Dropdown from '../Dropdown';

describe('Avatar component', () => {
  it('should render without issues', () => {
    const component = shallow(<Dropdown />);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});