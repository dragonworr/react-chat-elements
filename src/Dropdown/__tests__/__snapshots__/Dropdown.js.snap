// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component should render without issues 1`] = `
<div
  className="rce-dropdown default rce-dropdown-open__nortwest"
  style={
    Object {
      "left": 0,
      "top": 0,
    }
  }
>
  <ul />
</div>
`;
